src/transformers/trainer.py

```diff
@@ -2796,9 +2796,16 @@ def _maybe_log_save_evaluate(self, tr_loss, grad_norm, model, trial, epoch, igno
         metrics = None
         if self.control.should_evaluate:
             metrics = self._evaluate(trial, ignore_keys_for_eval)
+            new_best_metric = self._determine_best_metric(metrics=metrics, trial=trial)
+
+            if self.args.save_strategy == SaveStrategy.BEST:
+                if new_best_metric:
+                    self.control.should_save = True
+                else:
+                    self.control_should_save = False
```

Comment:
This can be refactored:

```suggestion
            self.control.should_save = new_best_metric
```